### PR TITLE
add types for react-native-file-type

### PR DESCRIPTION
**Pull request by InkAndSteel**
This pull request adds TypeScript type definitions for the react-native-file-type library. The type definitions include:

FileType function: An asynchronous function that accepts a file URI and returns a FileTypeResult containing the MIME type and file extension or null if not detected.

FileTypeResult interface: Defines the structure of the returned result, including:
- ext: The file extension (e.g., jpg).
- mime: The MIME type (e.g., image/jpeg).

### CODE REVIEW

File: types/react-native-file-type/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "rules": {
+        "@definitelytyped/no-any-union": "off"
```
Comment (rbuckton) [inline types/react-native-file-type/.eslintrc.json:3]: Why was this necessary?

Comment (jakebailey) [reply types/react-native-file-type/.eslintrc.json:3]: Certainly not.

**Review by jakebailey — CHANGES_REQUESTED** [review]
You need tests for your code, and please remove the eslintrc, which is doing nothing.

File: types/react-native-file-type/.eslintrc.json (hunk 1 above)
Comment (InkAndSteel) [reply types/react-native-file-type/.eslintrc.json]: is removed

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-native-file-type/react-native-file-type-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6cdfb367d6127b55ccf48cae60306b9ee9b49f1a
</p>

**Review by jakebailey — APPROVED** [review]
I'll OK this, but I will note that `react-native-file-type` is technically quite broken because it publishes an ESM `index.js` while not marking itself as `"type": "module"`. It's not clear this package would ever work outside of bundling. Use at your own risk.

https://publint.dev/react-native-file-type@1.0.0

**Comment by InkAndSteel** [comment]
> I'll OK this, but I will note that `react-native-file-type` is technically quite broken because it publishes an ESM `index.js` while not marking itself as `"type": "module"`. It's not clear this package would ever work outside of bundling. Use at your own risk.
> 
> https://publint.dev/react-native-file-type@1.0.0

Thx! :)

**Comment by InkAndSteel** [comment]
Ready to merge

**Review by sandersn — APPROVED** [review]

**Comment by InkAndSteel** [comment]
Ready to merge

**Pull request merged**